File: sdks/php/sdk/src/Documents/Requests/GenerateDocumentPreviewRequest.php
<?php

namespace Pogodoc\Documents\Requests;

use Pogodoc\Core\Json\JsonSerializableType;
use Pogodoc\Documents\Types\GenerateDocumentPreviewRequestType;
use Pogodoc\Core\Json\JsonProperty;
use Pogodoc\Core\Types\ArrayType;
use Pogodoc\Documents\Types\GenerateDocumentPreviewRequestFormatOpts;

class GenerateDocumentPreviewRequest extends JsonSerializableType
{
    /**
     * @var string $templateId
     */
    public string $templateId;

    /**
     * @var value-of<GenerateDocumentPreviewRequestType> $type
     */
    #[JsonProperty('type')]
    public string $type;

    /**
     * @var array<string, mixed> $data
     */
    #[JsonProperty('data'), ArrayType(['string' => 'mixed'])]
    public array $data;

    /**
     * @var ?GenerateDocumentPreviewRequestFormatOpts $formatOpts
     */
    #[JsonProperty('formatOpts')]
    public ?GenerateDocumentPreviewRequestFormatOpts $formatOpts;

    /**
     * @param array{
     *   templateId: string,
     *   type: value-of<GenerateDocumentPreviewRequestType>,
     *   data: array<string, mixed>,
     *   formatOpts?: ?GenerateDocumentPreviewRequestFormatOpts,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->templateId = $values['templateId'];
        $this->type = $values['type'];
        $this->data = $values['data'];
        $this->formatOpts = $values['formatOpts'] ?? null;
    }
}

File: sdks/csharp/sdk/src/PogodocApi/Documents/Requests/GenerateDocumentPreviewRequest.cs
using System.Text.Json.Serialization;
using PogodocApi.Core;

namespace PogodocApi;

public record GenerateDocumentPreviewRequest
{
    [JsonIgnore]
    public required string TemplateId { get; set; }

    [JsonPropertyName("type")]
    public required GenerateDocumentPreviewRequestType Type { get; set; }

    [JsonPropertyName("data")]
    public object Data { get; set; } = new Dictionary<string, object?>();

    [JsonPropertyName("formatOpts")]
    public GenerateDocumentPreviewRequestFormatOpts? FormatOpts { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
